### PR TITLE
[GAL-352] [GAL-349] Admire & Comment

### PR DESCRIPTION
**Pull request by tbezman**
### Changes
- All of the feed events throw an error if they are unable to render
  - No longer have to do a bunch of conditional rendering at the top level `FeedEvent`
  - Fallback is we just hide the feed event (it's like nothing ever happened)
- Removed background color from feed events when hovering
- Usernames are now all black and bold in the feed event headers.

### New stuff
The entire socialize section + the notes modal
image
image

### Demos
https://user-images.githubusercontent.com/6754223/195221350-c8d4265f-f68c-4518-8ac3-a02d626bc260.mov

https://user-images.githubusercontent.com/6754223/195222550-058ab5ae-65a3-4b7f-b388-df0f97b6492e.MOV

### CODE REVIEW

**Comment by tbezman** [comment]
@kaitoo1 @Robinnnnn I was having some trouble logging in on mobile ios in the vercel preview. hoping y'all can help check this out tomorrow.

File: src/components/Feed/Socialize/AdmireLine.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import { useFragment } from 'react-relay';
```
Comment (tbezman) [inline src/components/Feed/Socialize/AdmireLine.tsx:1]: this file is for this component 
image

File: src/components/Feed/Socialize/CommentBox.tsx [hunk 2]
```diff
@@ -0,0 +1,326 @@
+import styled, { css } from 'styled-components';
```
Comment (tbezman) [inline src/components/Feed/Socialize/CommentBox.tsx:1]: this file captures this entire component 
image

File: src/components/Feed/Socialize/CommentLine.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import { useFragment } from 'react-relay';
```
Comment (tbezman) [inline src/components/Feed/Socialize/CommentLine.tsx:1]: this file is for this component 
image

File: src/components/Feed/Socialize/FeedEventSocializeSection.tsx [hunk 4]
```diff
@@ -0,0 +1,267 @@
+import { useFragment } from 'react-relay';
```
Comment (tbezman) [inline src/components/Feed/Socialize/FeedEventSocializeSection.tsx:1]: this file captures this entire section 
image

File: src/components/Feed/Socialize/Interactions.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
+import { VStack } from 'components/core/Spacer/Stack';
```
Comment (tbezman) [inline src/components/Feed/Socialize/Interactions.tsx:1]: this file captures this section 
image

there's a good chunk of logic just to determine what combination of comments / admires to show

File: src/components/HoverCard/HoverCardOnUsername.tsx [hunk 6]
```diff
@@ -89,7 +89,9 @@ export default function HoverCardOnUsername({ userRef, queryRef }: Props) {
   return (
     <StyledContainer onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
       <StyledLinkContainer>
-        <InteractiveLink to={`/${user.username}`}>{user.username}</InteractiveLink>
+        <Link href={`/${user.username}`}>
+          <TitleDiatypeM>{user.username}</TitleDiatypeM>
+        </Link>
```
Comment (tbezman) [inline src/components/HoverCard/HoverCardOnUsername.tsx:94]: jess changed up how usernames get rendered in the feed event header 
image

File: src/components/Feed/Events/CollectionCreatedFeedEvent.tsx [hunk 7]
```diff
@@ -72,7 +72,7 @@ export default function CollectionCreatedFeedEvent({ eventRef, queryRef }: Props
   const collectionName = unescape(event.collection.name ?? '');
 
   if (!tokens.length || !event.owner) {
-    return null;
+    throw new Error('Tried to render CollectionCreatedFeedEvent without an owner / tokens');
```
Comment (Robinnnnn) [inline src/components/Feed/Events/CollectionCreatedFeedEvent.tsx:75]: thank you for doing this!

File: src/components/Feed/FeedList.tsx [hunk 8]
```diff
@@ -62,48 +67,60 @@ export default function FeedList({
...
+          parent={parent}
+        >
+          {({ registerChild }) => (
+            // @ts-expect-error: this is the suggested usage of registerChild
+            <div ref={registerChild} style={style}>
+              <FeedEvent
+                onPotentialLayoutShift={() => {
+                  measurerCache.clear(index, 0);
+                  virtualizedListRef.current?.recomputeRowHeights();
+                }}
```
[the remark this answers is not in the document]
Comment (Robinnnnn) [reply src/components/Feed/FeedList.tsx:111]: whenever a new comment is added, or a new notes, etc., we need to tell the virtualizer that something changed and re-compute the row

File: src/components/Feed/Socialize/CommentBox.tsx [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+          comment @prependNode(connections: $connections, edgeTypeName: "FeedEventCommentEdge") {
+            dbid
+            ...CommentLineFragment
+          }
+        }
+      }
+    }
+  `);
+
+  const [value, setValue] = useState('');
```
Comment (Robinnnnn) [inline src/components/Feed/Socialize/CommentBox.tsx:74]: leave comment on controlled input

File: src/components/Feed/Socialize/CommentBox.tsx [hunk 10]
```diff
@@ -0,0 +1,325 @@
...
+      setTimeout(() => {
+        onPotentialLayoutShift();
+      }, 100);
+
+      if (response.commentOnFeedEvent?.__typename === 'CommentOnFeedEventPayload') {
+        resetInputState();
+
+        onClose();
+      } else {
+        // error handle
```
Comment (Robinnnnn) [inline src/components/Feed/Socialize/CommentBox.tsx:148]: !

File: src/components/Feed/Socialize/CommentBox.tsx [hunk 11]
```diff
@@ -0,0 +1,325 @@
...
+    <>
+      {active && <ClickPreventingOverlay onClick={onClose} />}
+      <CommentBoxWrapper active={active}>
+        <Wrapper onClick={handleClick}>
+          <InputWrapper gap={12}>
+            {/* Purposely not using a controlled input here to avoid cursor jitter */}
+            <Textarea
+              onKeyDown={handleInputKeyDown}
+              ref={textareaRef}
+              onInput={(event) => {
```
Comment (Robinnnnn) [inline src/components/Feed/Socialize/CommentBox.tsx:195]: comment

File: src/components/Feed/Socialize/Interactions.tsx [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+
+      return (
+        <VStack gap={8}>
+          <AdmireLine admireRef={admire} queryRef={query} />
+
+          <RemainingAdmireCount remainingCount={remainingInteractions} eventRef={event} />
+        </VStack>
+      );
+    } else {
+      // link x admired this
```
Comment (Robinnnnn) [inline src/components/Feed/Socialize/Interactions.tsx:122]: ???

File: src/components/Feed/Socialize/NotesModal.tsx [hunk 13]
```diff
@@ -0,0 +1,229 @@
...
+
+  const [measurerCache] = useState(() => {
+    return new CellMeasurerCache({
+      fixedWidth: true,
+      minHeight: 0,
+    });
+  });
+
+  const handleSeeMore = useCallback(() => {
+    loadNext(10);
```
Comment (Robinnnnn) [inline src/components/Feed/Socialize/NotesModal.tsx:92]: use const please

**Pull request merged**